overlord/devicestate/handlers_install.go

```diff
@@ -312,6 +314,19 @@ func checkEncryption(model *asserts.Model) (res bool, err error) {
 		return false, nil
 	}
 
+	// check if kernel has fde-setup hook based encryption support
+	if kernelInfo, err := snapstate.KernelInfo(st, deviceCtx); err == nil {
+		if hasFDESetupHookInKernel(kernelInfo) {
+			// run the hook with "op":"features", only use hook
+			// if that is successful
+			err := checkFDEFeatures(st, kernelInfo)
+			if err == nil {
+				return true, nil
+			}
+			logger.Noticef("cannot use fde-setup hook: %v", err)
+		}
```

Comment:
we should return with false here, no?  it doesn't make sense I think to check for tpm if the kernel had a hook, or am I confused?

Reply:
Probably not, I was thinking it would do no harm but YAGNI and all that, it's easy enough to tweak this further. I added a comment.

Reply:
Actually, let me rework this first by merging master and the "storage-safety" work.

Reply:
This is now updated